R/add-subvariable.R

```diff
@@ -12,25 +13,55 @@
 #' }
 #' @export
 addSubvariable <- function (variable, subvariable) {
-    ## Get subvariable URL or URLs, depending on how many supplied
-    new.urls <- urls(subvariable)
+    subvariable <- add_subvar_def(variable, subvariable)
 
-    ## Store these for post workaround
-    subvar.urls <- subvariables(tuple(variable))
+    ## There is some inconsistency in the output order after patching the subvariable catalog
+    ## this stores the proper order.
+    order <- c(names(subvariables(variable)), vapply(subvariable, name, ""))
 
+    ## Get subvariable URL or URLs, depending on how many supplied
+    new.urls <- urls(subvariable)
     ## Do the adding
     crPATCH(shojiURL(variable, "catalogs", "subvariables"),
         body=toJSON(sapply(new.urls, emptyObject, simplify=FALSE)))
-
-    ## Workaround because apparently bind/rebind isn't retaining the order
-    crPATCH(self(variable),
-        body=toJSON(list(subvariables=I(c(subvar.urls, new.urls)))))
-
+    variable <- refresh(variable)
+    subvariables(variable) <- subvariables(variable)[order]
```

Comment:
Can you remove this update entirely? There have been changes on the backend since this workaround was put here, so it may not be necessary anymore. 

If it is still necessary, I think we should keep the existing implementation, even if it is less readable, because it is more efficient (saves several potentially large requests). The users that use this function use it heavily, so it's important that we don't make unnecessary requests.

Reply:
It looks like this is still necessary. When I take out the second patch it returns subvariables in the wrong order. I'll return it to the previous method,